### PR TITLE
Merge discovery

### PR DESCRIPTION
**Pull request by mjpitz**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -10,6 +12,7 @@ fmt:
 
 deps:
 	go mod download
+	go mod verify
```
Comment (MadhuvanthG) [inline Makefile:15]: Nice! Just looked up on what this is for.

File: hack/config/full.yaml [hunk 2]
```diff
@@ -0,0 +1,115 @@
+accounts:
```
Comment (MadhuvanthG) [inline hack/config/full.yaml:1]: How are these files generated? They look like document for how to structure a config, right?

File: internal/config/config.proto [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+    string per_page_parameter = 3;
+    string page_parameter = 4;
+    int32 page_size = 5;
+    string selector = 6;
+    Clone clone = 7;
+
+    Basic basic = 10;
+}
+
+message Static {
```
Comment (MadhuvanthG) [inline internal/config/config.proto:87]: Is this for supporting repos hosted outside of known platforms like Github, Bitbucket etc..?

File: hack/config/full.yaml (hunk 2 above)
Comment (mjpitz) [reply hack/config/full.yaml:1]: They aren't unfortunately. They're handwritten.

Comment (mjpitz) [reply hack/config/full.yaml:1]: And yes, the demonstrate some example config files.

File: internal/config/config.proto (hunk 3 above)
Comment (mjpitz) [reply internal/config/config.proto:87]: This is more for testing but useful to have around for other cases. The Generic one is really for systems outside of GitHub / GitLab. I came across CodeBerg which would be good to add support for.

**Pull request merged**